src/providers/execution/contracts/staking_router.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from src.utils.cache import global_lru_cache as lru_cache
+
+from web3.types import BlockIdentifier
+
+from src.providers.execution.base_interface import ContractInterface
+from src.utils.dataclass import list_of_dataclasses
+from src.web3py.extensions.lido_validators import StakingModule, NodeOperator
+from src.variables import EL_REQUESTS_BATCH_SIZE
+
+logger = logging.getLogger(__name__)
+
+
+class StakingRouterContract(ContractInterface):
```

Comment:
As suggestion, make StakingRouterContract an abstract class and then 
use StakingRouterContractV1 and StakingRouterContractV2 as implementation.

Currently things a little bit confusing with get_staking_modules which overridden in StakingRouterContractV2

Reply:
fixed